src/magic.h

```diff
@@ -202,6 +214,23 @@ class spell_type
 
         enum_bitset<spell_flag> spell_tags;
 
+        // sound related info
+        std::string s_msg;
+        int s_vol;
+        sounds::sound_t s_type;
+        std::string s_id;
+        std::string s_var;
+        bool s_emit;
+
+        // moral effect related info
+        std::set<std::string> m_senses;
+        morale_type m_type;
+        int m_bonus;
+        int m_bonus_rng_mod;
+        int m_max_bonus;
+        int m_duration;
```

Comment:
You are using this like it's a time duration, so why not make this a `time_duration`?

Reply:
my bad - the morale stuff will be done as a fake spell so this will be removed. The durations in the spell_type class and spell json's are generally handeld as int atm idK why.

Reply:
hold on a sec, morale stuff is new data, so it should go in spell_type. the durations in spell_type are saved in moves, and time_duration only has values as small as 1 second (100 moves)

fake_spell is for data that would be in the spell class: it's used because items and monsters don't have spells that level up, only the player (in magiclysm)

Reply:
ok.